scalafix-rules/src/main/scala/scala/tools/nsc/interactive/ScalafixGlobalProxy.scala

```diff
@@ -40,6 +41,7 @@ trait ScalafixGlobalProxy { this: ScalafixGlobal =>
           )
       }
     }
+    this.scheduler = new WorkScheduler
```

Comment:
So I guess I don't fully understand this part. What exactly does create and assigning a new WorkScheduler do? Mainly how does it help this situation?